common/src/main/java/com/graphicsfuzz/common/util/UpgradeShadingLanguageVersion.java

```diff
@@ -74,37 +86,87 @@
    */
   private List<Stmt> globalVariableInitializers = new ArrayList<>();
 
+  // Variable name renaming map. Maps original names with new names.
+  private Map<String, String> variableRename = new HashMap<>();
+
+  // Function name renaming map. Maps original names with new names.
+  private Map<String, String> functionRename = new HashMap<>();
+
+  /*
+   * Originally texture fetch functions included information about the kind of sampler
+   * to be used, but later on these were simplified and the remaining texture fetch
+   * functions work with various samplers.
+   */
```

Comment:
I am not keen on the use of a 2D array for what is intuitively a String-to-String map.

I think it would be better to use a Map<String, String> here, and either initialize it in the constructor via a series of "put" operations, or do:

final Map<String, String> textureFunctionRenames = new HashMap<String, String>() {{
        put("shadow1D", "texture");
        put("shadow2D", "texture");
        ...
    }};

Alternatively, define a method to do the renaming:

private String renameTextureFunction(String originalName) {
  switch (originalName) {
    case "shadow1D":
    case "shadow2D":
         ...
    case "textureCube":
      return "texture";
    // similar for other cases
}

Reply:
Done as a bunch of put calls in a separate function.